### DISCUSSION TITLE
Handling False-positives

### DISCUSSION THREAD

**Discussion by Mattk70** [Mattk70/Chirpity-Electron#34] [Show and tell]
I'll write this post to illustrate a workflow for tidying up BirdNET detections using Chirpity. Ultimately, you will end up with a clean archive of your records and the option to export a CSV that you can use as a basis for submissions to eBird, Trektellen, BirdTrack or whatever.

Here are the initial results of an analysis of one 3 hour file:

![image](https://github.com/Mattk70/Chirpity-Electron/assets/61826357/5df70185-d8fe-4e81-b017-bc121f824442)

The first thing I'll do is scan the summary table for unlikely species in my region, and delete them. For example, Savi's Warbler is a pretty rare vagrant, so these detections can be discarded:

![image](https://github.com/Mattk70/Chirpity-Electron/assets/61826357/9d0c3ece-da74-4c56-a33f-5236565cda82)

I'll repeat that process for any easily discarded species.

>[!note]
> If you want to suppress unlikely predictions, you can [use a custom list](https://github.com/Mattk70/Chirpity-Electron/discussions/394)

Now, I'll move on to species that _are_ possible.  When you select a species in the summary, Chirpity will jump to the first detection and highlight it. Pressing P will play that detection, so you can quickly ascertain if it's good. Tab will move to the next detection in the list. If the detection is wrong, you have two options:

1. Option 1: You can delete it (using the context menu, or just press either the "Delete" or "backspace" keys). If you delete a record Chirpity moves you to the next one. I often just hit delete repeatedly. Which leads me to this point:

- If you got a bit trigger-happy, you can also un-delete, with Ctrl or Command Z

2. Option 2: you can correct it. To do this right-click (or option click) the offending record, and pick "Edit Record"

![image](https://github.com/Mattk70/Chirpity-Electron/assets/61826357/c4d5ab8b-affa-45e9-9fa4-07fea86aebc4)

The species dropdown has focus when the pop-up opens, so you can immediately type the initial letters of the species until you get the right one, or, you can pick from the drop-down. As you can see, it's also possible to add a call count, label or comments.

When I am happy with this list. I can export it to CSV (File > Export results to CSV). If I just want to export the results from one species, I can filter by that species before export, but if the view is unfiltered, a file with all the detections will be created. That file is here:

[All_detections.csv](https://github.com/Mattk70/Chirpity-Electron/files/14123346/All_detections.csv)

You can also export audio clips of your detections - optionally with a couple of seconds padding and/or a fade effect. However, the view must be filtered to do this, Chirpity can only batch export audio for one species at a time.

**Comment by UlfGran** [comment]
Hi,

I have the following questions/comments:

In many cases, there are many duplicate detections of the most common/active species. In many hours of recording, there can be hundreds, or even thousands, of detections of the same species. In my case, I am not interested in multiple detections of the same species. I want a workflow where I can keep only one or a few detections of each species. My interest is to detect the presence of each species, not to keep hundreds of duplicate detections of the species.  Manually deleting hundreds of multiple detections is not feasible. I need a way of only keeping the detections I have selected for keeping, and reject the rest. Probably the easiest way of acheiving this would be to have the option to only save those detections that I have marked as "Reviewed" to the archive. Or, possibly, only the ones I have given a specific label, altough this takes more clicks and is less convenient. Is any of this possible? If not, I would like to see this implemented.

Or, can you suggest some other workflow that easily acheives my goal?

Regards
Ulf

**Reply by Mattk70** [reply]
Hi Ulf, 

You can verify results using the keyboard shortcut ctrl-v. If, having completed your review, you only wish to save verified results to the archive, set the confidence threshold to 100% before saving. This should ensure only verified results remain.
